### DISCUSSION TITLE
Differences between similar named flags

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#5064] [Q&A] [answered]
I see in profiles sometimes

caps
And
caps.drop all

But what is more secure? 

I see seccomp too

seccomp
seccomp.block-secondary

What is difference? 

Thanks and
best regards

**Comment by rusty-snake** [comment, marked as answer]
> **--caps**
> Linux capabilities is a kernel feature designed to split up the root privilege into a set of distinct  privileges. These  privileges can be enabled or disabled independently, thus restricting what a process running as root can do in the system.
>
> By default root programs run with all capabilities enabled. --caps option  disables  the  following  capabilities: CAP_SYS_MODULE,  CAP_SYS_RAWIO, CAP_SYS_BOOT, CAP_SYS_NICE, CAP_SYS_TTY_CONFIG, CAP_SYSLOG, CAP_MKNOD, CAP_SYS_ADMIN.  The filter is applied to all processes started in the sandbox.


> **--caps.drop=all**
> Drop all capabilities for the processes running in the sandbox. This option is recommended for  running  GUI  programs or any other program that doesn't require root privileges. It is a must-have option for sandboxing untrusted programs installed from unofficial sources - such as games, Java programs, etc.

Less capabilities = More secure. (_obvious_)

> **--seccomp**
> Enable seccomp filter and blacklist the syscalls in the default list, which is @default-nodebuggers  unless  --allow-debuggers is specified, then it is @default.
>
> System architecture is strictly imposed only if flag --seccomp.block-secondary is used. The filter is  applied  at run time only if the correct architecture was detected. For the case of I386 and AMD64 both 32-bit and 64-bit filters are installed.

> **--seccomp.block-secondary**
> Enable  seccomp  filter  and filter system call architectures so that only the native architecture is allowed. For example, on amd64, i386 and x32 system calls are blocked as well as changing the execution domain  with  personality(2) system call.

**Comment by osevan** [comment]
caps.drop=all = is more resrictive. 

Its true?

**Reply by rusty-snake** [reply]
`--caps` = drops a finite list of capabilities 
`--caps.drop=all` = drops ***ALL*** capabilities

`man 1 firejail` and `man 7 capabilities`.

**Reply by osevan** [reply]
Thank you very much
